awscli/customizations/s3/utils.py

```diff
@@ -217,10 +218,23 @@ def get_file_stat(path):
     """
     try:
         stats = os.stat(path)
-        update_time = datetime.fromtimestamp(stats.st_mtime, tzlocal())
-    except (ValueError, IOError) as e:
+    except IOError as e:
         raise ValueError('Could not retrieve file stat of "%s": %s' % (
             path, e))
+
+    try:
+        update_time = datetime.fromtimestamp(stats.st_mtime, tzlocal())
+    except ValueError as e:
+        # Python's fromtimestamp raises value errors when the timestamp is out
+        # of range of the platform's C localtime() function. This can cause
+        # issues when syncing from systems with a wide range of valid timestamps
+        # to systems with a lower range. Some systems support 64-bit timestamps,
+        # for instance, while others only support 32-bit. We set the time to be
+        # UNIX epoch time so that file size will take precedence in sync.
+        LOG.warning("Attempting to access file with out of range timestamp. "
```

Comment:
Do we need the LOG warnings anymore?  The result_queue will handle printing out error messages.